sentry-android-core/src/main/java/io/sentry/android/core/AnrV2EventProcessor.java

```diff
@@ -411,11 +420,59 @@ private void setPlatform(final @NotNull SentryBaseEvent event) {
     }
   }
 
-  private void setExceptions(final @NotNull SentryEvent event) {
-    final Throwable throwable = event.getThrowableMechanism();
-    if (throwable != null) {
-      event.setExceptions(sentryExceptionFactory.getSentryExceptions(throwable));
+  @Nullable
+  private SentryThread findMainThread(final @Nullable List<SentryThread> threads) {
+    if (threads != null) {
+      for (SentryThread thread : threads) {
+        final String name = thread.getName();
+        if (name != null && name.equals("main")) {
+          return thread;
+        }
+      }
+    }
+    return null;
+  }
+
+  // by default we assume that the ANR is foreground, unless abnormalMechanism is "anr_background"
+  private boolean isBackgroundAnr(final @NotNull Object hint) {
+    if (hint instanceof AbnormalExit) {
+      final String abnormalMechanism = ((AbnormalExit) hint).mechanism();
+      if (abnormalMechanism == null) {
+        return false;
+      }
+
+      if (abnormalMechanism.equals("anr_foreground")) {
+        return false;
+      }
+
+      return abnormalMechanism.equals("anr_background");
```

Comment:
`l`: I guess you added the options for readability right? Otherwise could be simplified to:
```suggestion
      final String abnormalMechanism = ((AbnormalExit) hint).mechanism();
      return "anr_background".equals(abnormalMechanism);
```